index.js

```diff
@@ -13,20 +13,18 @@ export class CancelError extends Error {
 
 export default class PCancelable {
 	static fn(userFunction) {
-		return (...arguments_) => {
-			return new PCancelable((resolve, reject, onCancel) => {
-				arguments_.push(onCancel);
-				// eslint-disable-next-line promise/prefer-await-to-then
-				userFunction(...arguments_).then(resolve, reject);
-			});
-		};
+		return (...arguments_) => new PCancelable((resolve, reject, onCancel) => {
+			arguments_.push(onCancel);
+			userFunction(...arguments_).then(resolve, reject);
+		});
 	}
 
 	constructor(executor) {
 		this._cancelHandlers = [];
 		this._isPending = true;
 		this._isCanceled = false;
 		this._rejectOnCancel = true;
+		this._isResolved = false;
```

Comment:
Outside the scope of this PR, but I think the logic could be simplified. Could be a follow-up PR if you're interested. Instead of using multiple variables, there's should simply be a `state` variable. A promise can only every be in the state `'pending' | 'resolved' | 'rejected' | 'canceled'`.

Reply:
> Could be a follow-up PR if you're interested.

Yes, I'm interested in it :)

Reply:
> Could be a follow-up PR if you're interested.

Yes, I'm interested in it :)

> Instead of using multiple variables, there's should simply be a state variable. 
> A promise can only every be in the state 'pending' | 'resolved' | 'rejected' | 'canceled'.

Understood. I will try to simplify the codes.